pkg/ui/console/policy.go

```diff
@@ -0,0 +1,163 @@
+		buildDescribePolicyTable(t, p.Policies, path.Join(policyPath, strings.ToLower(p.Name)))
+	}
+}
+
+func getNestedPolicyExample(p *policy.Policy, policyPath string) string {
+	if len(p.Policies) > 0 {
+		return getNestedPolicyExample(p.Policies[0], path.Join(policyPath, strings.ToLower(p.Name)))
+	}
+	return policyPath
+}
+
+func printPolicyResponse(results []*policy.ExecutionResult) {
+	if results == nil {
+		return
+	}
+	for _, execResult := range results {
+		ui.ColorizedOutput(ui.ColorUnderline, "%s %s Results:\n\n", emojiStatus[ui.StatusInfo], execResult.PolicyName)
+
+		if !execResult.Passed {
+			if execResult.Error != "" {
+				ui.ColorizedOutput(ui.ColorHeader, ui.ColorErrorBold.Sprintf("%s Policy failed to run\nError: %s\n\n", emojiStatus[ui.StatusError], execResult.Error))
+			} else {
+				ui.ColorizedOutput(ui.ColorHeader, ui.ColorErrorBold.Sprintf("%s Policy finished with warnings\n\n", emojiStatus[ui.StatusWarn]))
+			}
+		}
+		fmtString := defineResultColumnWidths(execResult.Results)
+		for _, res := range execResult.Results {
+			switch {
+			case res.Passed:
+				ui.ColorizedOutput(ui.ColorInfo, fmtString, emojiStatus[ui.StatusOK]+" ", res.Name, res.Description, color.GreenString("passed"))
+				ui.ColorizedOutput(ui.ColorInfo, "\n")
+			case res.Type == policy.ManualQuery:
+				ui.ColorizedOutput(ui.ColorInfo, fmtString, emojiStatus[ui.StatusWarn], res.Name, res.Description, color.YellowString("manual"))
+				ui.ColorizedOutput(ui.ColorInfo, "\n")
+				outputTable := createOutputTable(res)
+				for _, row := range strings.Split(outputTable, "\n") {
+					ui.ColorizedOutput(ui.ColorInfo, "\t\t  %-10s \n", row)
+				}
+
+			default:
+				ui.ColorizedOutput(ui.ColorInfo, fmtString, emojiStatus[ui.StatusError], res.Name, res.Description, color.RedString("failed"))
+				ui.ColorizedOutput(ui.ColorWarning, "\n")
+				queryOutput := findOutput(res.Columns, res.Data)
+				if len(queryOutput) > 0 {
+					for _, output := range queryOutput {
+						ui.ColorizedOutput(ui.ColorInfo, "\t\t%s  %-10s \n\n", emojiStatus[ui.StatusError], output)
+					}
+
+				}
+
+			}
+			ui.ColorizedOutput(ui.ColorWarning, "\n")
+		}
+	}
+}
+
+func findOutput(columnNames []string, data [][]interface{}) []string {
+	outputKeys := []string{"id", "identifier", "resource_identifier", "uid", "uuid", "arn"}
+	outputKey := ""
+	outputResources := make([]string, 0)
+	for _, key := range outputKeys {
+		for _, column := range columnNames {
+			if key == column {
+				outputKey = key
+			}
+		}
+		if outputKey != "" {
+			break
+		}
+	}
+	if outputKey == "" {
+		return []string{}
+	}
+	for index, column := range columnNames {
+		if column != outputKey {
+			continue
+		}
+		for _, row := range data {
+			outputResources = append(outputResources, fmt.Sprintf("%v", row[index]))
+		}
+	}
+	return outputResources
+}
+
+func createOutputTable(res *policy.QueryResult) string {
+	data := make([][]string, 0)
+	for rowIndex := range res.Data {
+		rowData := make([]string, 1)
```

Comment:
> We're doing an append below, so this would mean rowData[0] is always empty. FYI, probably wrong. Did you mean make([]string, 0, 1)?

(This is still here but moved to another file)

Reply:
That's what I said 😂

Reply:
🤫 I fix I fix